yt/utilities/on_demand_imports.py

```diff
@@ -37,167 +89,20 @@ def __init__(self, pkg_name):
             # yt-project/yt#1966
             self.error = ImportError(
                 "This functionality requires the %s "
-                "package to be installed." % self.pkg_name
+                "module to be installed." % self.pkg_name
```

Comment:
While we're at it, let's use f-strings!

Reply:
I intend to remove the `NotAModule` class in a later stage of this PR.
Normally the noun "package" is still relevant here because the error should trigger only on missing packages (which In consider a special class of modules).